packages/mongodb-chatbot-evaluation/src/evaluate/evaluateExpectedLinks.test.ts

```diff
@@ -0,0 +1,201 @@
+import { ObjectId } from "mongodb-rag-core";
+import { ConversationGeneratedData } from "../generate";
+import { evaluateExpectedLinks } from "./evaluateExpectedLinks";
+import { EvalResult } from "./EvaluationStore";
+
+const generateConversationData = (expectedLinks?: string[]) =>
+  ({
+    _id: new ObjectId(),
+    commandRunId: new ObjectId(),
+    type: "conversation",
+    data: {
+      _id: new ObjectId(),
+      createdAt: new Date(),
+      messages: [
+        {
+          createdAt: new Date(),
+          id: new ObjectId(),
+          role: "user",
+          content: "What's my name?",
+        },
+        {
+          createdAt: new Date(),
+          id: new ObjectId(),
+          role: "assistant",
+          content: "Your name is Jasper.",
+          references: [
+            {
+              title: "foo",
+              url: "bar",
+            },
+            {
+              title: "baz",
+              url: "fizz_buzz",
+            },
+          ],
+        },
+      ],
+    },
+    evalData: {
+      qualitativeFinalAssistantMessageExpectation:
+        "The assistant should correctly respond with the user's name.",
+      name: "User name",
+      expectedLinks,
+    },
+  } satisfies ConversationGeneratedData);
+describe("evaluateExpectedLinks", () => {
+  it("should return 1 when all expected links are found in the assistant message", async () => {
+    const generatedConversationData = generateConversationData(["bar", "fizz"]);
+    const runId = new ObjectId();
+    const evalResult = await evaluateExpectedLinks({
+      generatedData: generatedConversationData,
+      runId,
+    });
+    expect(evalResult).toMatchObject({
+      result: 1,
+      generatedDataId: generatedConversationData._id,
+      type: "expected_links",
+      commandRunMetadataId: runId,
+      createdAt: expect.any(Date),
+      _id: expect.any(ObjectId),
+      metadata: {
+        testName: generatedConversationData.evalData.name,
+        expectedLinksFound: {
+          bar: {
+            matchingActualLink: "bar",
+            includesExpected: true,
+          },
+          fizz: {
+            matchingActualLink: "fizz_buzz",
+            includesExpected: true,
+          },
+        },
```

Comment:
I think we could tweak the schema a bit here.

1. It's a bit weird to have links we expected but didn't find under `expectedLinksFound`
2. I think `includesExpected` is effectively the same as `matchingActualLink !== undefined`. If so we can make this implicit in the data model.

Maybe something like this?

```ts
interface Metadata {
  expectedLinks: {
    // Map of expectedLinks to one or more matching links we found
    matched: { [expectedLink: string]: [string, ...string[]] };
    // List of links we expected but didn't find
    unmatched: string[];
  };
}
```

So this test would be:

```ts
{
  metadata: {
    testName: generatedConversationData.evalData.name,
    expectedLinks: {
      matched: {
        bar: ["bar"],
        fizz: ["fizz_buzz"],
      },
      unmatched: [],
    },
  },
}
```

Reply:
seems reasonable. will adjust

Reply:
w/ a bit more thought on it, went with:

```ts
export interface ExpectedLinks {
  /**
    The expected link is the key and the value is an array
    of actual links that match the expected link.
   */
  [expectedLink: string]: string[];
}
```

and array is empty if no expected links. bit of a cleaner API